packages/nuxt/src/pages/module.ts

```diff
@@ -170,7 +170,21 @@ export default defineNuxtModule({
           if (nuxt.apps.default) {
             nuxt.apps.default.pages = pages
           }
+          const addedPagePaths = new Set<string>()
           function addPage (parent: EditableTreeNode, page: NuxtPage) {
+            // Avoid duplicate keys in the generated RouteNamedMap type
+            if (addedPagePaths.has(`${parent.path}${page.path}`)) {
+              if (page.meta) {
+                parent.addToMeta(page.meta)
+              }
+              if (page.alias) {
+                parent.addAlias(page.alias)
+              }
+              parent.name = page.name
+              return
+            }
+            addedPagePaths.add(`${parent.path}${page.path}`)
```

Comment:
should we use `joinURL` or `${parent.path}/${page.path}`?

Reply:
@danielroe 
I think it's preferable to use joinURL to avoid potential issues with path formatting. I've made the fix accordingly.